src/pages/PeoplePage.tsx

```diff
@@ -0,0 +1,125 @@
+import { useEffect, useMemo, useState } from 'react';
+import { Loader } from '../components/Loader';
+import { Person } from '../types';
+import { getPeople } from '../api';
+import { PeopleTable } from '../components/PeopleTable';
+import { PeopleFilters } from '../components/PeopleFilters';
+import { useSearchParams } from 'react-router-dom';
+
+export const PeoplePage: React.FC = () => {
+  const [people, setPeople] = useState<Person[]>([]);
+  const [hasError, setHasError] = useState(false);
+  const [isLoading, setIsLoading] = useState(true);
+
+  const [searchParams] = useSearchParams();
+
+  const sexStatus = searchParams.get('sex');
+  const query = searchParams.get('query');
+  const centuries = searchParams.getAll('centuries');
+  const sortType = searchParams.get('sort');
+  const sortOrder = searchParams.get('order');
+
+  useEffect(() => {
+    const fetchData = async () => {
+      try {
+        const data = await getPeople();
+
+        setPeople(data);
+      } catch {
+        setHasError(true);
+      } finally {
+        setIsLoading(false);
+      }
+    };
+
+    fetchData();
+  }, []);
+
+  const filteredPeople = useMemo(() => {
+    let result = [...people];
+
+    switch (sexStatus) {
+      case 'f':
+        result = result.filter(person => person.sex === 'f');
+        break;
+      case 'm':
+        result = result.filter(person => person.sex === 'm');
+        break;
+    }
+
+    if (query) {
+      result = result.filter(person => person.name.includes(query));
+    }
+
+    if (centuries.length) {
+      result = centuries.reduce<Person[]>((acc, c) => {
+        const filtered = result.filter(
+          person => +person.born.toString().slice(0, 2) === +c - 1,
+        );
+
+        return [...acc, ...filtered];
```

Comment:
The current implementation of the centuries filter may lead to duplicates in the result array if the same person belongs to multiple centuries. Consider using a different approach, for instance, checking if the person's century is included in the selected centuries array.